File: Data Structures & Algorithms/longest-increasing-subsequence/submission-3.py
class Solution:
    def lengthOfLIS(self, nums: List[int]) -> int:
        dp = [0]*2001
        for i in range(len(nums)):
            dp[i] = 1
            for j in range(i):
                if nums[j] < nums[i]:
                    dp[i] = max(dp[i], dp[j] + 1)
        return max(dp)

File: Data Structures & Algorithms/two-integer-sum/submission-3.go
func twoSum(nums []int, target int) []int {
    m := make(map[int]int) // value -> index
    for i, v := range nums {
        complement := target - v
        if j, ok := m[complement]; ok {
            return []int{j, i}
        }
        m[v] = i
    }
    return nil
}